cider/include/cider/CiderCompileModule.h

```diff
@@ -50,10 +51,11 @@ class CiderCompilationResult {
 
 class CiderCompileModule {
  public:
-  CiderCompileModule();
+  CiderCompileModule(std::shared_ptr<CiderAllocator> allocator);
```

Comment:
should provide a default value? or in `Make` method?

Reply:
Actually we can't, we expect that when using this module, we must pass in the allocator to ensure the consistency of the entire allocation process. Of course, we just currently provide a CiderDefaultAllocator in the test unit.

Reply:
it make sense. And we need to do same thing for `CiderRuntimeModule`, you can do that in next pr.